src/parser-babylon.js

```diff
@@ -25,12 +26,14 @@ function parse(text) {
     ]
   };
 
+  const parseMethod = options.parseExpression ? "parseExpression" : "parse";
```

Comment:
Oh, [`parseJavaScriptExpression`](https://github.com/prettier/prettier/blob/b7e1c366e0088a338eecc43f6e8691302966e8fe/src/multiparser.js#L228-L236)? I could try to reuse it, but it would be a little tricky, since `babylon.parse` parses `"string"` parses as a `Directive` rather than a `StringLIteral`. Maybe it would make more sense to have `parseJavaScriptExpression` also use `babylon.parseExpression`.